ddl/serial_test.go

```diff
@@ -591,7 +591,10 @@ func (s *testSerialSuite) TestCancelAddIndexPanic(c *C) {
 	}
 	c.Assert(checkErr, IsNil)
 	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[ddl:8214]Cancelled DDL job")
+	errMsg := err.Error()
+	// Cancelling the job can either succeed or not, it depends on whether the cancelled job takes affect.
+	// For now, there's no way to guarantee that cancelling will always take effect.
```

Comment:
better add a todo here, when issue  #17904 is closed, there is no need to tolerate it here.

Reply:
Please address the comment @djshow832